resources/vaft-ublock-origin.js

```diff
@@ -108,35 +82,35 @@
                 ${adRecordgqlPacket.toString()}
                 ${tryNotifyTwitch.toString()}
                 ${parseAttributes.toString()}
-                declareOptions(self);
-                self.TwitchAdblockSettings = ${JSON.stringify(TwitchAdblockSettings)};
-                self.addEventListener('message', function(e) {
-                    if (e.data.key == 'UpdateIsSquadStream') {
-                        IsSquadStream = e.data.value;
-                    } else if (e.data.key == 'UpdateClientVersion') {
-                        ClientVersion = e.data.value;
-                    } else if (e.data.key == 'UpdateClientSession') {
-                        ClientSession = e.data.value;
-                    } else if (e.data.key == 'UpdateClientId') {
-                        ClientID = e.data.value;
-                    } else if (e.data.key == 'UpdateDeviceId') {
-                        GQLDeviceID = e.data.value;
-                    } else if (e.data.key == 'UpdateClientIntegrityHeader') {
-                        ClientIntegrityHeader = e.data.value;
-                    } else if (e.data.key == 'UpdateAuthorizationHeader') {
-                        AuthorizationHeader = e.data.value;
-                    }
-                });
-                hookWorkerFetch();
-                importScripts('${jsURL}');
+                ${getWasmWorkerUrl.toString()}
+                var workerUrl = getWasmWorkerUrl('${twitchBlobUrl}');
```

Comment:
`${getWasmWorkerUrl.toString()}` is a constant defined within this script (it prints the source of the `getWasmWorkerUrl` function like all the other `${}` parts.

`${twitchBlobUrl}` however is a variable passed in to the overridden `new Worker(twitchBlobUrl)` constructor.

A top-level script calling `new Worker("'+console.log(123)+'")` (i.e. with a "URL" like `'+console.log(123)+'`) will execute `console.log(123)` within the context of the Worker.

It's difficult to see a reason why this would be useful to an adversary, as they either already have scripting or can somehow alter the URL argument to the Worker constructor. However, I'd feel more comfortable with 

```suggestion
                var workerUrl = getWasmWorkerUrl('${twitchBlobUrl.replace("'", "%27")}');
```

Reply:
cc @pixeltris

Reply:
Added sanity checks in https://github.com/pixeltris/TwitchAdSolutions/commit/39666c7bb1446b3d5ac711d43e6b2ce5a5e54b2f